src/vs/base/common/search.ts

```diff
@@ -21,3 +25,20 @@ export function buildReplaceStringWithCasePreserved(matches: string[] | null, pa
 		return pattern;
 	}
 }
+
+function validateSpecificSpecialCharacter(matches: string[], pattern: string, specialCharacter: string): boolean {
+	const doesConatinSpecialCharacter = matches[0].indexOf(specialCharacter) !== -1 && pattern.indexOf(specialCharacter) !== -1;
+	const doesConatinSameNumberOfSpecialCharacters = matches[0].split(specialCharacter).length === pattern.split(specialCharacter).length;
```

Comment:
Could we fast return if `doesConatinSpecialCharacter` is false to avoid uncessary `split` cost. For example:

```
return doesConatinSpecialCharacter && matches[0].split(specialCharacter).length === pattern.split(specialCharacter).length;
```

Reply:
Makes perfect sense , Thanks for this :)